core/types/calls.cc

```diff
@@ -271,12 +268,22 @@ SymbolRef guessOverload(const GlobalState &gs, SymbolRef inClass, SymbolRef prim
 
                 auto argType = Types::resultTypeAsSeenFrom(gs, candidate.data(gs)->arguments()[i].type,
                                                            candidate.data(gs)->owner, inClass, targs);
-                if (argType->isFullyDefined() && !Types::isSubType(gs, arg->type, argType)) {
+                if (argType->isFullyDefined() && !Types::isSubType(gs, arg, argType)) {
                     it = leftCandidates.erase(it);
                     continue;
                 }
                 ++it;
             }
+        };
+
+        // Lets see if we can filter them out using arguments.
+        for (auto i = 0; i < numPosArgs; ++i) {
+            checkArg(i, args[i]->type);
+        }
+
+        // If keyword args are present, interpret them as an untyped hash
+        if (numPosArgs < args.size()) {
+            checkArg(numPosArgs, Types::hashOfUntyped());
```

Comment:
A good follow-up to this PR would be to fix this case to use the keyword arguments during overload checking. right now, it treats keyword arguments present in the send as being an untyped hash in the last argument.

Reply:
#37